crates/networking/discv5/src/discovery.rs

```diff
@@ -61,15 +72,14 @@ struct QueryResult {
 
 pub struct Discovery {
     discv5: Discv5,
-    local_enr: Enr,
     event_stream: EventStream,
     discovery_queries: FuturesUnordered<Pin<Box<dyn Future<Output = QueryResult> + Send>>>,
     find_peer_active: bool,
     pub started: bool,
 }
 
 impl Discovery {
-    pub async fn new(local_key: Keypair, config: &DiscoveryConfig) -> anyhow::Result<Self> {
+    pub async fn new(local_key: Keypair, config: &DiscoveryConfig) -> anyhow::Result<(Self, Enr)> {
```

Comment:
```suggestion
    pub async fn new(local_key: Keypair, config: &DiscoveryConfig) -> anyhow::Result<Self> {
```

instead of returning the ENR from new we should just
let discovery = Discovery::new(x,y,z);
let enr = discovery.local_enr();

local_enr() would fetch it from discv5